js/map.js

```diff
@@ -134,21 +138,65 @@
       photosElementContainer.appendChild(newImgElement);
     });
 
-
     return newCardElement;
   }
 
   function renderCard(data, elementContainer) {
     elementContainer.insertBefore(data, elementContainer.firstElementChild);
   }
 
-  mapElement.classList.remove('map--faded');
+  function mouseUpHendler() {
+    mapElement.classList.remove('map--faded');
+    adFormContainer.classList.remove('ad-form--disabled');
+    updateAdress(getPinLocation(mapElement, '.map__pin--main', PIN_HIGHT, HALF_PIN_WIDTH), formAddressInput);
+    renderPins(generatePins(pinElementTemplate, nearbyLisitings), mapPinsElement);
+  }
+
+  function updateAdress(position, element) {
+    element.value = position.top + ', ' + position.left;
+  }
+
+  function getPinLocation(container, pinName, pinHight, halfPinWidth) {
+    var pinContainerElement = container.querySelector(pinName);
+    return {
+      top: (parseFloat(getComputedStyle(pinContainerElement).top) + parseFloat(getComputedStyle(pinContainerElement, 'after').top) + pinHight),
+      left: (parseFloat(getComputedStyle(pinContainerElement).left) + parseFloat(getComputedStyle(pinContainerElement, 'after').left) + halfPinWidth)
+    };
+  }
+
+  function showCard(elem, arr) {
+    renderCard(generateCard(arr[+elem.dataset.index], cardElementTemplate), filterElementContainer);
+  }
+
+  function removeCard() {
+    filterElementContainer.querySelector('.map__card').remove();
+  }
+
+  function pinsHandler(event) {
+    var button = event.target.parentElement;
```

Comment:
bug you code only works when you click on an image, if you click on the border it won't open the popup.

Reply:
your code should be along this lines:
```
while (target != mapElement) {
    if (target.tagName == 'BUTTON') {
      removeCard();
      showCard(button, nearbyLisitings);
      return;
    }
target = target.parentNode;
}

or while (target != this) { ...} as this is the current target, the listener attached to.
    
```
This is an alternative to closest

Reply:
done